services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/TransferManagerFactory.java

```diff
@@ -56,6 +57,10 @@ public static S3TransferManager createTransferManager(DefaultBuilder tmBuilder)
             log.warn(() -> "The provided DefaultS3AsyncClient is not an instance of S3CrtAsyncClient, and thus multipart"
                            + " upload/download feature is not enabled and resumable file upload is not supported. To benefit "
                            + "from maximum throughput, consider using S3AsyncClient.crtBuilder().build() instead.");
+        } else if (s3AsyncClient instanceof MultipartS3AsyncClient) {
+            log.warn(() -> "The provided S3AsyncClient is an instance of MultipartS3AsyncClient, and thus multipart"
```

Comment:
Didn't we add tests for this in https://github.com/aws/aws-sdk-java-v2/pull/4903/

Reply:
#4903 was merged in the feature branch not master